analysis.go

```diff
@@ -111,142 +52,142 @@ func GetAnalysisTasks(tasks chan<- Task, basepath string, projects []string, res
 
 	// project specific analysis in here.
 	for _, p := range projects {
-		JSONResourceFactory := getDumpedJSONResourceFactory(filepath.Join(basepath, "projects", p))
-		tasks <- CheckProjectTask(p, results, JSONResourceFactory)
+		tasks <- CheckProjectTask(basepath, p, results)
 	}
 }
 
-// CheckProjectTask is a task factory for tasks that diagnose system conditions.
-func CheckProjectTask(project string, results chan<- CheckResults, JSONResourceFactory DumpedJSONResourceFactory) Task {
-	return checkProjectTask(func() []CheckTask {
-		return []CheckTask{CheckEventLogForErrors, CheckDeployConfigsReplicasNotZero, CheckForWaitingPods}
-	}, JSONResourceFactory, project, results)
-}
-
-// A getProjectCheckFactory generates tasks to diagnose system conditions.
-type getProjectCheckFactory func() []CheckTask
-
-// CheckResults Stores the result of a check in a scope which can a specfic
-// project or platform-wide.
-type CheckResults struct {
-	Scope   string `json:"scope"`
-	Name    string `json:"name,omitempty"`
-	Results []Result
-}
-
-// checkTasks executes all the CheckTasks returned from the supplied
-// checkFactory against the specified project. The results of the checks are
-// combined into a single JSON object and returned.
-func checkProjectTask(checkFactory getProjectCheckFactory, JSONResourceFactory DumpedJSONResourceFactory, project string, results chan<- CheckResults) Task {
+// CheckProjectTask returns a task that diagnoses problems in the project scope.
+func CheckProjectTask(basepath, project string, results chan<- CheckResults) Task {
 	return func() error {
-		result := CheckResults{Scope: "project", Name: project, Results: []Result{}}
-		checks := checkFactory()
-
-		var errors errorList
-		for _, check := range checks {
-			res, err := check(JSONResourceFactory)
-			if err != nil {
-				errors = append(errors, err)
-			}
-			result.Results = append(result.Results, res)
+		result := CheckResults{
+			Project: project,
+		}
 
+		var (
+			events            types.EventList
+			deploymentConfigs types.DeploymentConfigList
+			pods              types.PodList
+		)
+
+		definition := definitionLoader{basepath: basepath, project: project}
+
+		definition.Load("events", &events)
+		definition.Load("deploymentconfigs", &deploymentConfigs)
+		definition.Load("pods", &pods)
+
+		if err := definition.Err(); err != nil {
+			// REVIEW: do we want / is there a clean way to have
+			// analysisErrorReadingDumpedResource status in the
+			// output? Alternatively, returning an error will get it
+			// into dump.log and stdout.
```

Comment:
@philbrookes we need to review the fate of `analysisErrorReadingDumpedResource`. Now that we load resources independent of running a `Check*`, there is no result where to put that error.

I think I need to further change this, because with the current implementation the task aborts on error loading any of the resources, but we want to attempt running each check independently.